### PR TITLE
Update german language &c.

### PR DESCRIPTION
**Pull request by Ludy87**
- update german language
- Menu Map 0-255 to 0-100 percent Fan speed
- unused function
- pins correction

### CODE REVIEW

File: Marlin/src/lcd/language/language_de.h [hunk 1]
```diff
@@ -200,9 +200,9 @@
 #define MSG_MOVE_E                          _UxGT("Bewege Extruder")
 #define MSG_HOTEND_TOO_COLD                 _UxGT("Hotend zu kalt")
 #define MSG_MOVE_0025MM                     _UxGT(" 0,025 mm")
-#define MSG_MOVE_01MM                       _UxGT(" 0,1 mm")
-#define MSG_MOVE_1MM                        _UxGT(" 1,0 mm")
-#define MSG_MOVE_10MM                       _UxGT("10,0 mm")
+#define MSG_MOVE_01MM                       _UxGT(" 0,100 mm")
+#define MSG_MOVE_1MM                        _UxGT(" 1,000 mm")
+#define MSG_MOVE_10MM                       _UxGT("10,000 mm")
 #define MSG_SPEED                           _UxGT("Geschw.")
```
Comment (AnHardt) [inline Marlin/src/lcd/language/language_de.h:206]: Do you have more than 1000 steps/mm? This suggests an exactness there isn't. If you just want to align the "mm" replace the zeros with spaces.

File: Marlin/src/lcd/menu/menu.cpp [hunk 2]
```diff
@@ -448,7 +448,7 @@ bool ui_selection; // = false
 void set_ui_selection(const bool sel) { ui_selection = sel; }
 void do_select_screen(PGM_P const yes, PGM_P const no, selectFunc_t yesFunc, selectFunc_t noFunc, PGM_P const pref, const char * const string/*=nullptr*/, PGM_P const suff/*=nullptr*/) {
   if (ui.encoderPosition) {
-    ui_selection = (ENCODERBASE) > 0 == int16_t(ui.encoderPosition) > 0;
+    ui_selection = ((ENCODERBASE) > 0) == (int16_t(ui.encoderPosition) > 0);
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu.cpp:451]: Ah right. `==` has higher/equal precedence to `>`.

File: Marlin/src/lcd/menu/menu_configuration.cpp [hunk 3]
```diff
@@ -303,7 +303,7 @@ static void lcd_factory_settings() {
     #endif
     START_MENU();
     MENU_BACK(MSG_CONFIGURATION);
-    MENU_ITEM_EDIT(uint8, MSG_FAN_SPEED, &ui.preheat_fan_speed[material], 0, 255);
+    MENU_ITEM_EDIT(percent, MSG_FAN_SPEED, &ui.preheat_fan_speed[material], 0, 255);
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu_configuration.cpp]: 0 - 255 … percent?

**Comment by thinkyhead** [comment]
### & etc.
The ampersand is a ligament of the Latin word "Et".
So one should write `&c.` or `etc.`, but `& etc.` is redundant.

**Comment by Ludy87** [comment]
in German the combination `& etc.` is not uncommon

**Pull request merged**